projects/swimlane/ngx-ui/src/lib/components/drawer/drawer.component.spec.ts

```diff
@@ -17,10 +18,80 @@ describe('DrawerComponent', () => {
   beforeEach(() => {
     fixture = TestBed.createComponent(DrawerComponent);
     component = fixture.componentInstance;
+
+    component.size = 10;
+    component.zIndex = 10;
+    component.closeOnOutsideClick = true;
+
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
+    expect(component.size).toEqual(10);
+    expect(component.zIndex).toEqual(10);
+    expect(component.closeOnOutsideClick).toEqual(true);
+  });
+
+  describe('cssClasses', () => {
+    it('should get left drawer classes', () => {
+      component.direction = DrawerDirection.Left;
+      expect(component.cssClasses).toContain('left-drawer');
+    });
+
+    it('should get bottom drawer classes', () => {
+      component.direction = DrawerDirection.Bottom;
+      expect(component.cssClasses).toContain('bottom-drawer');
+    });
+  });
+
+  describe('onEscapeKey', () => {
+    it('should close', () => {
+      const spy = spyOn(component.close, 'emit');
+      component.onEscapeKey();
+      expect(spy).toHaveBeenCalled();
+    });
+  });
+
+  describe('setDimensions', () => {
+    it('should set size when direction left', done => {
+      component.direction = DrawerDirection.Left;
+      component.setDimensions(10);
+
+      setTimeout(() => {
```

Comment:
why a timeout here? would `fixture.detectChanges` and `fixture.whenStable` work better here?